src/gt4py/next/program_processors/runners/dace_iterator/__init__.py

```diff
@@ -184,11 +184,9 @@ def get_cache_id(
     return m.hexdigest()
 
 
-def run_dace_iterator(program: itir.FencilDefinition, *args, **kwargs) -> None:
+def build_sdfg_from_itir(program: itir.FencilDefinition, *args, **kwargs) -> dace.SDFG:
```

Comment:
What do you think about changing the signature of this function and explicitly passing the parameters that were already extracted by the caller?
`auto_optimize: bool`
`device: dace.DeviceType` (we can skip run_on_gpu)
`column_axis: Optional[Dimension]`
`offset_provider: dict[str, Any]`

Reply:
@edopao Yes this sounds like a good idea. Could you review the changes, please.